File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:

        finalArray=[]

        for i in range(numRows):
            rowArray=[]
            maxLength=i
            
            for j in range(maxLength+1):

                if j==0 or j==maxLength or i==0:
                    rowArray.append(1)
                else:
                    rowArray.append(finalArray[i-1][j]+finalArray[i-1][j-1])
                
            finalArray.append(rowArray)
            print(finalArray)

        return finalArray

        

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result=new ArrayList<>();
        for(int i=1;i<=numRows;i++){
            int ans=1;
            List<Integer> temp=new ArrayList<>();
            temp.add(1);
            for(int j=1;j<i;j++){
                ans*=(i-j);
                ans/=j;
                temp.add(ans);
            }
            result.add(new ArrayList<>(temp));
        }
        return result;
    }
}